### ISSUE TITLE
Twisted should use @decorators

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4630]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)| @jyknight reported|
|-|-|
|Trac ID|trac#4630|
|Type|task|
|Created|2010-08-05 20:33:37Z|
|Branch|https://github.com/twisted/twisted/tree/decorator-docs-4630|

Instead of avoiding them everywhere, it should instead use them where they make sense. Now that all supported versions support them, there's no reason not to.

classmethod, inlineCallbacks, and deprecated are but a few of the function decorators that would be better spelled with the prefix @decorator syntax.

Attachments:

* [decorator-doc.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f1c/f1c162e8edb528b0bf34c56e74499b0d8e0cc3d2/972a2f053f4c0620a161dbb4de137abbdc36caf4.patch) (3341 bytes) - added by frap on 2010-09-09 19:34:27Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4630 4630
type__task task
reporter__jknight jknight
priority__low low
milestone__ 
branch__branches_decorator_docs_4630 branches/decorator-docs-4630
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1281040417000000 1281040417000000
changetime__1284835880000000 1284835880000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

This ticket is far too broad.  Tickets should have well defined resolution conditions, with narrowness of scope preferred over things which encompass all of Twisted.  I think it should be closed as invalid.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

Replying to [exarkun](#note_1):
> This ticket is far too broad.  Tickets should have well defined resolution conditions, with narrowness of scope preferred over things which encompass all of Twisted.  I think it should be closed as invalid.

I was going to say that we should more clearly define its scope as "the coding standard should be updated to allow usage of `@decorator` syntax", but then [I actually read the coding standard](http://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html) and noticed that it doesn't actually forbid it; it was just implicitly forbidden by our support of Python 2.3.

I don't think we need to explicitly add any text to say that you _can_ use any particular syntactic feature, so if we can all agree that the coding standard is correct in failing to prohibit it, we don't actually need to do anything and I agree we should close it as invalid.

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_3"></a>|@jyknight commented|
|-|-|

Okay. If that's agreed, I'd repurpose this ticket to a more specific goal:

The documentation for t.i.defer.inlineCallbacks, t.i.defer.deferredGenerator, and t.p.deprecate should document use as @decorator rather than as after-def decoration.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

> The documentation for t.i.defer.inlineCallbacks, t.i.defer.deferredGenerator, and t.p.deprecate should document use as @decorator rather than as after-def decoration.

Sure.

**Comment by twisted-trac** [comment]
|<img alt="frap's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|frap set owner to frap<br>frap set status to `assigned`|
|-|-|

Ill try my hand at this simple doc fix

**Comment by twisted-trac** [comment]
|<img alt="frap's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|frap removed owner<br>frap set status to `new`|
|-|-|

patch updates  t.i.defer.inlineCallbacks, t.i.defer.deferredGenerator, and t.p.deprecate with @decorate syntax - just easy documentation fix

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

This coding standard change, as it sounds to me, should also be include in the development documentation.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [30041]) Branching to 'decorator-docs-4630'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [30042]) Apply decorator-doc.patch 

refs [#4630](https://github.com/twisted/twisted/issues/4630)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @exarkun|
|-|-|

Since the coding standard doesn't say anything about decorators, I don't think there's anything we need to change in it.

Otherwise, change looks good to me.  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [30044]) Merge decorator-docs-4630

Author: frap
Reviewer: exarkun
Fixes: [#4630](https://github.com/twisted/twisted/issues/4630)

Change the documentation examples for `inlineCallbacks`,
`deferredGenerator`, and `deprecated` to use decorator
syntax.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
